src/components/withWindowDimensions.js

```diff
@@ -16,7 +16,15 @@ const windowDimensionsPropTypes = {
 };
 
 export default function (WrappedComponent) {
-    class withWindowDimensions extends Component {
+    const propTypes = {
+        forwardedRef: PropTypes.func,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+    };
+
+    class WithWindowDimensions extends Component {
```

Comment:
Should be camelCase I think

Reply:
I actually had to change this to `PascalCase` so that it can be rendered like `<WithWindowDimensions />`. I guess if you try and do `<withWindowDimensions />`, react will think you're trying to render regular html like `<input>` or `<table>` or whatever and complain that it doesn't support the `<withWindowDimensions />` tag. Maybe I was just holding something wrong, but changing it to `PascalCase` fixed the error.

Anyways, the good news is that it uses a default export so it can still be imported and used in camelCase.

Reply:
Huh, how strange.

Maybe we should change our code style to capitalize components. All of the ones in the react native docs are PascalCase...

Reply:
I wonder why it was working before but not now. Can you account for why that discrepancy is occurring?

Reply:
Short answer:

Before, we weren't ever explicitly rendering it like `<withWindowDimensions />`, but now we _are_ explicitly rendering it on line 86.

---

Long Answer:

`withWindowDimensions` as you know is a HOC, which means that it is a function that takes a component as input and returns a component. You'll notice that the `WrappedComponent` is using `PascalCase`, and is rendered starting on line 69.

Before, we weren't doing any explicit rendering of the `withWindowDimensions` component itself, but rather calling it as a function, getting a component back, and rendering that.

We are able to pass props through `withWindowDimensions` to the `WrappedComponent` by simply drilling them down (as done on line 62 before or line 71 now). But refs work differently. In order to pass a ref through to a wrapped component, we need to use `React.forwardRef`, so what we're doing now is something like this:

`React.forwardRef(withWindowDimesions(WrappedComponent))`

But in order to forward the ref through `withWindowDimensions`, `withWindowDimensions` needs to accept the ref as a prop and then pass it as a ref to the `WrappedComponent`. So that's why we're explicitly rendering `<WithWindowDimensions />` on line 86 - so that we can pass the forwarded ref as a prop.

---

Caveat: I admit this feels a bit weird but I can't think of any other way to pass a ref through a HOC to a wrapped component. Open to any ideas if there's an easier way to do this.

Reference for where I found this solution: https://dev.to/justincy/using-react-forwardref-and-an-hoc-on-the-same-component-455m

Reply:
> Maybe we should change our code style to capitalize components

Well, that is normally how we do this, except all our HOCs use `camelCase` by convention. This still uses `camelCase` everywhere else in the code and works just the same, except it can now accept a ref and pass it through.

Reply:
Ah ok, thanks for this explanation! Very nice ❤️

Reply:
Just passing through, but I agree with Luke that this should follow the camel case convention like other HOC. 

My 2 cents, it would have been simpler to just drill the `withWindowDimensions` props down through `EmojiPickerMenu`. Instead we increased the complexity of a very general thing to cover a use case that we will very rarely run into.